lib/pushmi_pullyu/aip_creator.rb

```diff
@@ -0,0 +1,73 @@
+require 'pushmi_pullyu'
+require 'pushmi_pullyu/aip_downloader'
+require 'bagit'
+require 'archive/tar/minitar'
+
+class PushmiPullyu::AipCreator
+
+  attr_reader :noid, :config, :logger,
+              :workdir, :basedir, :tar_filename
+
+  def initialize(noid, config = nil, logger = nil)
+    @noid = noid
+
+    @config = config || PushmiPullyu.options
+    @logger = logger || PushmiPullyu.logger
+
+    @workdir = File.expand_path(self.config[:workdir])
+    @basedir = "#{workdir}/#{noid}"
+    @tar_filename = "#{workdir}/#{noid}.tar"
+  end
+
+  def create_aip(skip_download: false, clean_work_directories: true)
+    download_aip unless skip_download
+    bag_aip
+    tar_bag
+    destroy_aip_directory if clean_work_directories
+  end
+
+  def cleanup
+    destroy_aip_directory
+    destroy_aip_file
+  end
+
+  def destroy_aip_file
+    FileUtils.rm(tar_filename) if File.exist?(tar_filename)
+  end
+
+  def destroy_aip_directory
+    aip_downloader.clean_directories
+  end
```

Comment:
I agree with this.  

You could enhance `create_aip` even to have it return the path to the tar file (call `tar_filename`/`aip_filename` at the very end of that method)? 

Then when a someone is accessing your stuff in CLI.rb `run_tick_loop` method or wherever it will just be:
```
aip_tar_file = PushmiPullyu::AipCreator.new(noid, options, logger).create_aip
#do swift stuff => SwiftUpload.new(aip_tar_file)?
```